### PR TITLE
Feature/peacock quiz

### PR DESCRIPTION
**Pull request by NUO97**
Merge into develop

### CODE REVIEW

File: perceptia-one/perceptia/src/components/QuestionView.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            completed: false,
+
+        }
+    }
+
+    componentDidMount() {
+
+        const { selectedQuiz } = this.props.location.state
+
+        fetch(constants.localhost.url + 'questions/' + selectedQuiz)
```
Comment (spirrochet) [inline perceptia-one/perceptia/src/components/QuestionView.js]: See comment in constants.js about using the api.url constant instead.

Comment (spirrochet) [inline perceptia-one/perceptia/src/components/QuestionView.js]: It does not look like there is any error handling for 4xx and 5xx http status codes, this should be handled here.

File: perceptia-one/perceptia/src/components/QuizGallery.js [hunk 2]
```diff
@@ -34,7 +39,8 @@ class QuizGallery extends React.Component {
     }
 
     componentDidMount() {
-        fetch('http://students.washington.edu/long27km/any_quiz/v1/list')
+    
+        fetch(constants.localhost.url + 'list')
```
Comment (spirrochet) [inline perceptia-one/perceptia/src/components/QuizGallery.js]: See comment in constants.js about using the api.url constant.

File: perceptia-one/perceptia/src/components/QuizInfo.js [hunk 3]
```diff
@@ -41,11 +44,12 @@ class QuizInfo extends React.Component {
   }
 
   componentDidMount() {
-      fetch('http://students.washington.edu/long27km/any_quiz/v1/read/' + this.state.quiz)
+      fetch(constants.localhost.url + 'read/' + this.state.quiz)
```
Comment (spirrochet) [inline perceptia-one/perceptia/src/components/QuizInfo.js]: See comment in constants.js about using the api.url constant.

**Review by spirrochet — CHANGES_REQUESTED** [review]
Basic review completed.

Note, no testing completed on code in branch, just visual inspection of source.

Issues: 

*  Using hard coded api route. Change to using api.url route. 

*  azure-pipelines.yml version for perceptiaone has not been updated.

**Review by spirrochet — APPROVED** [review]

File: perceptia-one/perceptia/src/components/constants.js [hunk 4]
```diff
@@ -13,6 +13,10 @@ export default {
 
     api: {
         url: api_ref
+    },
+
+    localhost: {
```
[the remark this answers is not in the document]
Comment (NUO97) [reply perceptia-one/perceptia/src/components/constants.js]: Fixed

File: perceptia-one/perceptia/src/components/QuestionView.js (hunk 1 above)
Comment (NUO97) [reply perceptia-one/perceptia/src/components/QuestionView.js]: Fixed

File: perceptia-one/perceptia/src/components/QuizGallery.js (hunk 2 above)
Comment (NUO97) [reply perceptia-one/perceptia/src/components/QuizGallery.js]: Fixed

File: perceptia-one/perceptia/src/components/QuizInfo.js (hunk 3 above)
Comment (NUO97) [reply perceptia-one/perceptia/src/components/QuizInfo.js]: Fixed

**Pull request merged**